File: universal-ai-artifacts/clients/python/examples.py
from universal_ai_client import UniversalAIClient
import os

BASE_URL = os.environ.get('BASE_URL', 'https://your-domain.com')
TOKEN = os.environ.get('TOKEN', 'YOUR_TOKEN')

client = UniversalAIClient(BASE_URL, TOKEN)

print('Chat:')
print(client.chat('Hello, how are you?', optimizePrompt=True))

print('\nStreaming:')
for event in client.stream_chat('Tell me a short story.'):
    if event.get('type') == 'chunk':
        print(event.get('content', ''), end='', flush=True)
    elif event.get('type') == 'end':
        print('\n[stream end]', event.get('metadata'))
        break

print('\nRAG Search:')
search = client.rag_search('machine learning algorithms', collection='knowledge_base', limit=3)
print([{'id': r['id'], 'score': r['score']} for r in search.get('results', [])])

print('\nPlugin Execute:')
print(client.execute_plugin('web-scraper', 'scrape', { 'url': 'https://example.com', 'selector': '.content', 'format': 'text' }))

File: universal-ai-artifacts/clients/node/src/examples.js
import UniversalAIClient from './client.js';

const BASE_URL = process.env.BASE_URL || 'https://your-domain.com';
const TOKEN = process.env.TOKEN || 'YOUR_TOKEN';

const client = new UniversalAIClient(BASE_URL, TOKEN);

async function main() {
  console.log('Chat:');
  const chat = await client.chat('Hello, how are you?', { optimizePrompt: true });
  console.log(chat);

  console.log('\nStreaming:');
  await client.streamChat('Tell me a short story.', (chunk) => {
    if (chunk.type === 'chunk') process.stdout.write(chunk.content);
    if (chunk.type === 'end') console.log('\n[stream end]', chunk.metadata);
  });

  console.log('\nRAG Search:');
  const search = await client.ragSearch('machine learning algorithms', { collection: 'knowledge_base', limit: 3 });
  console.log(search.results?.map(r => ({ id: r.id, score: r.score })));

  console.log('\nPlugin Execute:');
  const plugin = await client.executePlugin('web-scraper', 'scrape', { url: 'https://example.com', selector: '.content', format: 'text' });
  console.log(plugin);
}

main().catch((err) => {
  console.error(err);
  process.exit(1);
});